src/tpm.rs

```diff
@@ -61,22 +62,43 @@ pub enum Prescale {
     Div128 = 0b111,
 }
 
-pub struct Tpm {
-    _reg: &'static mut TPMRegs,
+pub enum ChannelSel {
+    Ch0 = 0,
+    Ch1 = 1,
+    Ch2 = 2,
+    Ch3 = 3,
+    Ch4 = 4,
+    Ch5 = 5,
 }
 
-impl Tpm {
-    // TODO: allow the selection of the channel
+//based on sourcing current PLL CLK w/ Div8 (48MHz)
+pub enum Positions {
+	FullyRetracted = 0x500,
+	Middle = 0x1000,
+	FullyActuated = 0x1E00,
+}
+
+pub struct Tpm<'a> {
```

Comment:
You might want to store the channel that you are controlling with this struct in here as well. Alternatively, we might want to have the `Tpm` struct take care of TPM global values and allow creating separate `Channel` structs from the `Tpm` that allow handling of individual channel values.